robozonky-strategy-natural/src/main/java/com/github/robozonky/strategy/natural/conditions/RevenueRateCondition.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 The RoboZonky Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.robozonky.strategy.natural.conditions;
+
+import java.math.BigDecimal;
+
+import com.github.robozonky.strategy.natural.Wrapper;
+
+public class RevenueRateCondition extends AbstractRangeCondition {
+
+    private static final BigDecimal MAX_RATE = BigDecimal.valueOf(Double.MAX_VALUE);
+
+    public RevenueRateCondition(final BigDecimal fromInclusive, final BigDecimal toInclusive) {
+        super(Wrapper::getRevenueRate, fromInclusive, toInclusive);
+        RevenueRateCondition.assertIsInRange(fromInclusive);
+        RevenueRateCondition.assertIsInRange(toInclusive);
+    }
+
+    public RevenueRateCondition(final BigDecimal fromInclusive) {
+        this(fromInclusive, RevenueRateCondition.MAX_RATE);
+    }
+
+    private static void assertIsInRange(final BigDecimal interestRate) {
```

Comment:
@triceo I could use some help about what constitutes a valid value. This looks like you're ok with any positive Double value. But the test strategy makes it look like 0 - 100 % are valid. What values should I allow in the web tool?

The same question applies to LoanAnnuity (what values should I allow) :)

Reply:
Yeah, I'm not really sure why there's no <0; 100> limit in here, but there definitely should be.
When it comes to `LoanAnnuity`, that is potentially unlimited - in practice, max would be somewhere around 20,000 CZK.